### DISCUSSION TITLE
Lane detectin -carla- is any othber simmulatin avilabel?

### DISCUSSION THREAD

**Discussion by jayajkd** [thomasfermi/Algorithms-for-Automated-Driving#35] [General]
Hello, good morning. I've already run the code you provided, and after completing all 20 epochs, it generated model directories. Actually, I'm working on line detection for autonomous vehicles on my college campus. I need to create a custom dataset to use with the model. Specifically, I'm looking for guidance on three aspects:

How can I generate my custom dataset for training?
What should be the format of the labels for this dataset?
Are there alternative methods or tools for data collection and simulation other than using the Carla simulator? because installing Carla on a large configuration system can be challenging.

**Comment by MankaranSingh** [comment]
You can read the code provided with book to understand the format for data. 
There are other simulations like MS air sim you can probably use for data collection.

**Comment by thomasfermi** [comment]
Hi @jayajkd ! 
Regarding the custom dataset: How does the track look like on campus? Is it similar to a public road? Then you could just use a model someone else trained on regular roads and published online.
Why are you interested in training in a simulator? If you want to use your model in the real world a simulator can only help you for pretraining. You will probably have to at least fine-tune on real data.

And a comment: If you want to use your model in the real world, you should really care about calibration. There is a chapter on that in the book.
